Fabric.Identity.API/scripts/Install-IdPSS-Utilities.psm1

```diff
@@ -0,0 +1,235 @@
+    if($null -eq $credential) {
+        $credential = New-AzureADApplicationPasswordCredential -ObjectId $objectId
+    }
+
+    return $credential.KeyId
+}
+
+function Connect-AzureADTenant {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $tenantId,
+        [Parameter(Mandatory=$true)]
+        [PSCredential] $credential
+    )
+
+    try {
+        Connect-AzureAD -Credential $credential -TenantId $tenantId | Out-Null
+    }
+    catch {
+        Write-DosMessage -Level "Error" -Message  "Could not sign into tenant '$tenantId' with user '$($credential.UserName)'"
+        throw
+    }
+}
+
+function Add-InstallationTenantSettings {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $tenantId,
+        [Parameter(Mandatory=$true)]
+        [string] $clientSecret,
+        [Parameter(Mandatory=$true)]
+        [string] $clientId,
+        [ValidateScript({
+            if (!(Test-Path $_)) {
+                throw "Path $_ does not exist. Please enter valid path to the install.config."
+            }
+            if (!(Test-Path $_ -PathType Leaf)) {
+                throw "Path $_ is not a file. Please enter a valid path to the install.config."
+            }
+            return $true
+        })]  
+        [string] $installConfigPath = "$(Get-CurrentScriptDirectory)\install.config"
+    )
+
+    $installationConfig = [xml](Get-Content $installConfigPath)
+    $tenantScope = $installationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $tenantSettings = $tenantScope.SelectSingleNode('tenants')
+
+    # Add a tenant section if not exists
+    if($null -eq $tenantSettings) {
+        $tenantSettings = $installationConfig.CreateElement("tenants")
+        $tenantScope.AppendChild($tenantSettings) | Out-Null
+    }
+
+    $existingSetting = $tenantSettings.ChildNodes | Where-Object {$_.tenantId -eq $tenantId}
+
+    if ($null -eq $existingSetting) {
+        $setting = $installationConfig.CreateElement("variable")
+
+        $nameAttribute = $installationConfig.CreateAttribute("tenantId")
+        $nameAttribute.Value = $tenantId
+        $setting.Attributes.Append($nameAttribute) | Out-Null
+
+        $valueAttribute = $installationConfig.CreateAttribute("secret")
+        $valueAttribute.Value = $clientSecret
+        $setting.Attributes.Append($valueAttribute) | Out-Null
+
+        $clientAttribute = $installationConfig.CreateAttribute("clientid")
+        $clientAttribute.Value = $clientId
+        $setting.Attributes.Append($clientAttribute) | Out-Null
+
+        $tenantSettings.AppendChild($setting) | Out-Null
+    }
+    else{
+        $existingSetting.secret = $clientSecret
+        $existingSetting.clientId = $clientId
+    }
+    $installationConfig.Save("$installConfigPath") | Out-Null
+}
+
+function Set-AppSettings($appDirectory, $appSettings){
```

Comment:
A similar Set-EnvironmentVariable/Add-EnvironmentVariable exists fabric-install-utilities for these two, but they go into slightly different locations (appSettings vs environmentVariables) with different encapsulating elements. Not sure if this should also go into the fabric-install-utilities eventually.